File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/remote/retrofit/ResponseTypeConverter.java
package com.zuk0.gaijinsmash.riderz.data.remote.retrofit;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/*
    This class provides custom annotations for Converter.Factory
*/
public final class ResponseTypeConverter {
    @Retention(RUNTIME)
    @interface Json {
    }

    @Retention(RUNTIME)
    @interface Xml {
    }
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/remote/retrofit/ResponseTypeConverter.kt
package com.zuk0.gaijinsmash.riderz.data.remote.retrofit

/*
    This class provides custom annotations for Converter.Factory
*/
class ResponseTypeConverter {
    @kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
    internal annotation class Json

    @kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
    internal annotation class Xml
}